spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -468,4 +436,549 @@ public void testAggregateExpressions() {
     assertThat(r1.get(10)).isEqualTo(0.25); // VAR_POP(num1)
     assertThat(r1.get(11)).isEqualTo(0.5); // VAR_SAMP(num1)
   }
+
+  @Test
+  public void testInnerJoin() {
+    writeTestDataToBigQuery(initialData(), testDataset.toString() + "." + testTable);
+    Dataset<Row> df = readTestDataFromBigQuery(testDataset.toString() + "." + testTable);
+
+    // Creating a data frame of schema NumStruct, and writing it to BigQuery to perform the join
+    // using pushdown
+    writeTestDataToBigQuery(
+        additionalData(), testDataset.toString() + "." + testTable + "_to_join");
+    Dataset<Row> df_to_join =
+        readTestDataFromBigQuery(testDataset.toString() + "." + testTable + "_to_join");
+
+    List<Row> result =
+        df.join(df_to_join, df.col("num1").equalTo(df_to_join.col("num1"))).collectAsList();
+    /*
+     +----+----+----+--------------------+----+----+----+--------------------+
+     |num1|num2|num3|             strings|num1|num2|num3|             strings|
+     +----+----+----+--------------------+----+----+----+--------------------+
+     |   4|   3|   2|[[[[3:str1, 4:str...|   4|   1|   3|[[[[1:str1, 4:str...|
+     |   3|   2|   1|[[[[2:str1, 3:str...|   3|   4|   3|[[[[4:str4, 3:str...|
+     +----+----+----+--------------------+----+----+----+--------------------+
+    */
+    assertThat(result.size()).isEqualTo(2);
+    for (Row r : result) {
+      assertThat(r.get(0)).isEqualTo(r.get(4));
+    }
+    result = df_to_join.join(df, df.col("num1").equalTo(df_to_join.col("num1"))).collectAsList();
+    /*
+     +----+----+----+--------------------+----+----+----+--------------------+
+     |num1|num2|num3|             strings|num1|num2|num3|             strings|
+     +----+----+----+--------------------+----+----+----+--------------------+
+     |   4|   1|   3|[[[[1:str1, 4:str...|   4|   3|   2|[[[[3:str1, 4:str...|
+     |   3|   4|   3|[[[[4:str4, 3:str...|   3|   2|   1|[[[[2:str1, 3:str...|
+     +----+----+----+--------------------+----+----+----+--------------------+
+    */
+    assertThat(result.size()).isEqualTo(2);
+    for (Row r : result) {
+      assertThat(r.size()).isEqualTo(8);
+      assertThat(r.get(0)).isEqualTo(r.get(4));
+    }
+
+    // disabling pushdown and testing the same
+    BigQueryConnectorUtils.disablePushdownSession(spark);
+    result = df.join(df_to_join, df.col("num1").equalTo(df_to_join.col("num1"))).collectAsList();
```

Comment:
Why do we override the result variable here? I thought here we could do something like 
assert(resultWithPushdown == resultWithoutPushdown)